Makefile.toml

```diff
@@ -614,6 +614,46 @@ docker run --rm \
 '''
 ]
 
+# Provide build options for netdog
+[tasks.netdog-tools]
+dependencies = ["setup", "fetch-sources"]
+script = [
+  '''
+  cargo install \
+    ${CARGO_MAKE_CARGO_ARGS} \
+    --path sources/api/netdog \
+    --root tools \
+    --force \
+    --quiet
+  '''
+]
+
+[tasks.netdog-parse]
```

Comment:
I'm wondering if this belongs in the Makefile.toml. Since it's a developer workflow anyway it could be done locally `cargo run --package net-dog -- parse --file blah`

Is it in the Makefile so that users (i.e. not just developers) can use it? If so, is there documentation missing?

Reply:
Yes, there is documentation missing, I actually was writing that this morning since I realized I forgot that. I mostly wanted to see if using a docker container this way was acceptable or if we wanted to plumb through providing the file directly and let it run natively like you suggest. We have similar workflows for bootconfig which made me think it was desirable to put this in the Makefile.

Reply:
> which made me think it was desirable to put this in the Makefile.

You may be right, let's see if anyone else chimes in. I can see this being useful to metal users. And if so, then having it in the Makefile simplifies things for them somewhat.

Reply:
> Since it's a developer workflow anyway it could be done locally cargo run --package net-dog -- parse --file blah

@webern, are you saying that creating a `net.toml` configuration file is a developer workflow? If that's the case, I disagree, I think it is both a developer and user workflow, since as @yeazelm said, a user shouldn't have to boot an instance in order to check if the configuration is valid, e.g. I don't want to wait until I boot my instance to know that I have a typo.

Reply:
A `cargo install` option might be a nice alternative to including this in the SDK -- you'd be able to avoid pulling down the container image if you hadn't already.

Reply:
I pulled this out of the PR. It sounds like we probably need to provide something different for folks that don't have the repo pulled and rust set up. Nonetheless, the new doc line works on different architectures much better, so one should be able to use their local machine (like a Mac) to run this command which I think might actually be an improvement until we solve the tools problem more broadly.